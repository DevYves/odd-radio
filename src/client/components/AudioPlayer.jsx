import React, {Component} from 'react';

class AudioPlayer extends Component {
  constructor(props) {
    super(props);
    this.state = {
      currentStationName: '',
      stream: '',
      streamType: ''
    }
    this.audioPlayer = this.audioPlayer.bind(this)
  }

  componentDidMount() {
    this.setState({
      currentStationName: this.props.stationFeed.name,
      stream: this.props.stationFeed.stream,
      streamType: this.props.stationFeed.type
    })
  }

  audioPlayer(event) {
    console.log(event);
  }

  render() {
    console.log('inside AudioPlayer');
    return (
      <div>
        <div className="player-station-name"> { this.props.currentStationName } </div>
        <audio controls="controls" >
          <source src={ this.state.stream } type={ this.state.streamType } />
        </audio>
        <div className="seek-button"><button>seek</button></div>
      </div>
    );
  }
}
export default AudioPlayer;