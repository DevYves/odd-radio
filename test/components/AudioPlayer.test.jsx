import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

// React 16 Enzyme adapter
Enzyme.configure({ adapter: new Adapter() });

import AudioPlayer from '../../src/client/components/AudioPlayer';

describe('AudioPlayer', () => {
  const props = {
    stationFeed: {
      type: 'audio/mpeg',
      stream: 'www.streamlink.ca',
      id: '1',
      name: 'Jest Test Stream'
    },
    seekStation: jest.fn()
  }

  const subject = <AudioPlayer { ...props }/>;

  it('matches snapshot', () => {
    expect(renderer.create(subject)).toMatchSnapshot();
  });

  describe('onClick', () => {
    const wrapper = shallow(subject);

    it('calls the function sent in by props', () => {
      wrapper.find('#pButton').simulate('click');
    });
  });
});