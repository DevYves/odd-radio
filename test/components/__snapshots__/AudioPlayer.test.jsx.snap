// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AudioPlayer matches snapshot 1`] = `
<div>
  <audio
    id="player"
    onCanPlay={[Function]}
    onLoadStart={[Function]}
  >
    <source
      src={undefined}
      type={undefined}
    />
  </audio>
  <div>
    <div
      className="audio-player-station-name"
      onClick={[Function]}
    >
       
      Jest Test Stream
       
    </div>
    <div
      className="container player-container"
    >
      <div
        className="row player-row border"
      >
        <div
          className="four columns"
        >
          <div>
            <i
              className="fa fa-play fa-2x audio-player-play-button"
              onClick={[Function]}
            />
          </div>
        </div>
        <div
          className="four columns"
        >
          <div>
            <div>
              <i
                className="fa fa-volume-up fa-2x"
                id="mute"
                onClick={[Function]}
              />
            </div>
            <input
              id="vol-control"
              max={100}
              min={0}
              onChange={[Function]}
              onInput={[Function]}
              step={1}
              type="range"
              value={100}
            />
          </div>
        </div>
        <div
          className="four columns"
        >
          <i
            className="fa fa-random fa-2x seek-button"
            onClick={[Function]}
          />
        </div>
      </div>
    </div>
    <div
      className="container presets"
    >
      <div
        className="center presets-title"
      >
        Presets: 
      </div>
      <div
        className="row border"
      >
        <div
          className="five columns border-right audio-player-presets "
          onClick={[Function]}
        >
          <p>
            1: 
            Empty
          </p>
        </div>
        <div
          className="five columns border-right audio-player-presets "
          onClick={[Function]}
        >
          <p>
            2: 
            Empty
          </p>
        </div>
        <div
          className="five columns border-right audio-player-presets "
          onClick={[Function]}
        >
          <p>
            3: 
            Empty
          </p>
        </div>
        <div
          className="five columns border-right audio-player-presets "
          onClick={[Function]}
        >
          <p>
            4: 
            Empty
          </p>
        </div>
        <div
          className="five columns audio-player-presets "
          onClick={[Function]}
        >
          <p>
            5: 
            Empty
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
